### PR TITLE
refactor: execute money transfers util in TestLoop

### PR DESCRIPTION
**Pull request by Longarithm**
This is a common logic used in all (3) TestLoop tests and I extract it to separate utils module. I want to achieve test logic to be as simple as possible.

### CODE REVIEW

File: integration-tests/src/test_loop/tests/in_memory_tries.rs [hunk 1]
```diff
@@ -87,43 +85,7 @@ fn test_load_memtrie_after_empty_chunks() {
...
-                transaction: tx,
-                is_forwarded: false,
-                check_only: false,
-            });
-        drop(future);
-    }
-
-    // Give plenty of time for these transactions to complete.
-    test_loop.run_for(Duration::seconds(40));
+    crate::test_loop::utils::execute_money_transfers(&mut test_loop, &node_datas, &accounts);
```
Comment (wacban) [inline integration-tests/src/test_loop/tests/in_memory_tries.rs:88]: nit: How about importing it? I don't think there is any risk of ambiguity given it takes test loop as the first param.

File: integration-tests/src/test_loop/utils.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+use near_primitives::types::AccountId;
+use std::collections::HashMap;
+
+pub(crate) const ONE_NEAR: u128 = 1_000_000_000_000_000_000_000_000;
+
+/// Execute money transfers within given `TestLoop` between given accounts.
+/// Used to generate state changes and check that chain is able to update
+/// balances correctly.
+pub(crate) fn execute_money_transfers(
+    test_loop: &mut near_async::test_loop::TestLoopV2,
```
Comment (wacban) [inline integration-tests/src/test_loop/utils.rs:18]: nit: How about importing it?

File: integration-tests/src/test_loop/utils.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    node_datas: &[TestData],
+    accounts: &[AccountId],
+) {
+    let clients = node_datas
+        .iter()
+        .map(|data| &test_loop.data.get(&data.client_sender.actor_handle()).client)
+        .collect_vec();
```
Comment (wacban) [inline integration-tests/src/test_loop/utils.rs:25]: This is also quite common, would it make sense to have this in a helper method? Or maybe wrap `node_datas` in a struct and add some typical usages as methods. Totally optional as this PR is about something else.

File: integration-tests/src/test_loop/utils.rs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    let mut balances = accounts
+        .iter()
+        .cloned()
+        .map(|account| (account.clone(), clients.query_balance(&account)))
+        .collect::<HashMap<_, _>>();
+    let num_clients = clients.len();
+
+    // Assume that all clients are in sync up to this block.
```
Comment (wacban) [inline integration-tests/src/test_loop/utils.rs:33]: nit: If this is important maybe assert that this is true?

File: integration-tests/src/test_loop/utils.rs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    let anchor_hash = clients[0].chain.head().unwrap().prev_block_hash;
+    drop(clients);
+
+    for i in 0..accounts.len() {
+        let amount = ONE_NEAR * (i as u128 + 1);
+        let tx = SignedTransaction::send_money(
+            1,
+            accounts[i].clone(),
+            accounts[(i + 1) % accounts.len()].clone(),
```
Comment (wacban) [inline integration-tests/src/test_loop/utils.rs:42]: nit: Those are used twice, maybe use variables like `sender` and `receiver` (or whatever those are called in the `send_money`).

**Review by wacban — APPROVED** [review]
LGTM

File: integration-tests/src/test_loop/utils.rs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+use std::collections::HashMap;
+
+pub(crate) const ONE_NEAR: u128 = 1_000_000_000_000_000_000_000_000;
+
+/// Execute money transfers within given `TestLoop` between given accounts.
+/// Used to generate state changes and check that chain is able to update
+/// balances correctly.
+pub(crate) fn execute_money_transfers(
+    test_loop: &mut near_async::test_loop::TestLoopV2,
+    node_datas: &[TestData],
```
Comment (tayfunelmas) [inline integration-tests/src/test_loop/utils.rs:19]: very nit. node_data

File: integration-tests/src/test_loop/utils.rs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+use near_client::test_utils::test_loop::ClientQueries;
+use near_network::client::ProcessTxRequest;
+use near_primitives::test_utils::create_user_test_signer;
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::types::AccountId;
+use std::collections::HashMap;
+
+pub(crate) const ONE_NEAR: u128 = 1_000_000_000_000_000_000_000_000;
+
+/// Execute money transfers within given `TestLoop` between given accounts.
```
Comment (tayfunelmas) [inline integration-tests/src/test_loop/utils.rs:14]: also add comment saying this also runs the chain long enough for the money transfers to be completed/finalized.

**Review by tayfunelmas — APPROVED** [review]

File: integration-tests/src/test_loop/utils.rs (hunk 3 above)
Comment (Longarithm) [reply integration-tests/src/test_loop/utils.rs:25]: I want to do something like that next, even more globally: I think we shouldn't deconstruct `TestLoopEnv` just as it is created, all these helpers could be methods of TestLoopEnv. But I also don't want to do this without Robin or Shreyan approval.

File: integration-tests/src/test_loop/utils.rs (hunk 4 above)
Comment (Longarithm) [reply integration-tests/src/test_loop/utils.rs:33]: Well then I'll just take minimal of all heights :)

**Pull request merged**